index.bs

```diff
@@ -305,35 +305,76 @@ For the main <dfn>sanitize</dfn> operation, using a {{ParentNode}} |node|, a
          exists, and if
          |config|["{{SanitizerConfig/elements}}"][|elementName|]["{{SanitizerElementNamespaceWithAttributes/removeAttributes}}"]
          [=SanitizerConfig/contains=] |attrName|:
-         1. Remove |attr| from |child|.
-      1. If &laquo;[|elementName|, |attrName|]&raquo; matches an entry in the
-         [=navigating URL attributes list=], and if |attr|'s [=protocol=] is
-         "`javascript:`":
-         1. Then remove |attr| from |child|.
+         1. Then [=remove an attribute|remove=] |attr|.
+      1. If the [=navigating URL attributes list=] [=SanitizerConfig/contains=]
+         &laquo;[|elementName|, |attrName|]&raquo; and |attr|
+         [=contains a javascript: URL=]:
+         1. Then [=remove an attribute|remove=] |attr|.
+      1. If |child|'s [=Element/namespace=] [=string/is=] the
+         [=MathML Namespace=] and |attr|'s [=Attr/local name=] [=string/is=]
+         "`href`" and  |attr| [=contains a javascript: URL=]:
+         1. Then [=remove an attribute|remove=] |attr|.
+      1. If the [=animating URL attributes list=] [=SanitizerConfig/contains=]
+         &laquo;[|elementName|, |attrName|]&raquo; and |attr|'s
+         [=get an attribute value|value=] [=string/is=] "`href`" or
+         "`xlink:href`":
+         1. Then [=remove an attribute|remove=] |attr|.
        1. Call [=sanitize=] on |child|'s [=Element/shadow root=] with |config|.
     1. else:
-      1. [=/remove=] |child|.
+      1. [=/Remove=] |child|.
 
 </div>
 
+<div class=note>
+<span class=marker>Note:</span> Current browsers support `javascript:` URLs
+only when navigating. Since navigation itself is not an XSS threat we treat
+navigation to `javascript:` URLs, but not navigations in general.
+
+Declarative navigation falls into a handful of categories:
+
+  1. Anchor elements. (`<a>` in HTML and SVG namespaces)
+  1. Form elements that trigger navigation as part of the form action.
+  1. MathML allows [any element to act as an anchor](
+     https://www.w3.org/TR/MathML3/mathml.html#chapter2_fund.globatt).
+  1. SVG animation.
+
+The first two are covered by the [=navigating URL attributes list=]. The MathML
+case is covered by a seperate rule, because there is no formalism in this spec
+to cover a "per-namespace global" rule. The animation case is covered by the
+[=animating URL attributes list=]. But since the interpretation of SVG animation elements depends on the animation target, and since during sanitization we cannot know what the final target will be, the [=sanitize=] algorithm blocks any animation of `href` attributes.
+
+</div>
+
+<div algorithm>
+To determine whether an |attribute| <dfn>contains a javascript: URL</dfn>, do this:
+
+1. Let |url| be the result of running the [=basic URL parser=]
+   on |attribute|'s [=get an attribute value|value=].
+1. Let |contains javascript: url| be:
+   |url| is not `failure` and |url|'s [=url/scheme=] [=string/is=] "`javascript`".
```

Comment:
I recommend just returning early if url is failure. I think that would be clearer.